### PR TITLE
unit tests for service.go in overlay package added

### PR DESCRIPTION
**Pull request by crawter**
unit tests for service.go in overlay package added

---
This change is [Reviewable](https://reviewable.io/reviews/storj/storj/155)

### CODE REVIEW

File: pkg/overlay/service_test.go [hunk 1]
```diff
@@ -224,3 +224,90 @@ func (o *MockOverlay) FindStorageNodes(ctx context.Context, req *proto.FindStora
...
+
+// 	defer os.RemoveAll(tempPath)
+
+// 	address := "127.0.0.1:15550"
+
+// 	client, err := NewTLSClient(&address, *opts, nil)
+
+// 	assert.NoError(t, err)
+// 	assert.NotNil(t, client)
+// }
```
Comment (nfarah86) [inline pkg/overlay/service_test.go]: do you want to remove the commented lines from the pr?

File: pkg/overlay/service_test.go [hunk 2]
```diff
@@ -224,3 +224,90 @@ func (o *MockOverlay) FindStorageNodes(ctx context.Context, req *proto.FindStora
...
+
+	result := service.InstanceID()
+
+	assert.NotNil(t, result)
+}
+
+func newTLSFileOptions(t *testing.T) (*peertls.TLSFileOptions, string) {
+	tempPath, err := ioutil.TempDir("", "TestNewPeerTLS")
+	assert.NoError(t, err)
+	//defer os.RemoveAll(tempPath)
```
Comment (nfarah86) [inline pkg/overlay/service_test.go]: may be remove this comment too?

**Comment by nfarah86** [comment]
i think outside of the commented line, it lgtm

**Comment by crawter** [comment]
I want to add a few comments according to my changes:
1. In file  pkg/kademlia/config.go
i have made this type public
type CtxKey int 
to be able reach this type in my tests. 

2. To successfully create RedisOverlayCache i need valid conn string to redis server. So i run command
"redis-server" and start server on 127.0.0.1 with default port 6379. I did it in next way:

	cmd := exec.Command("redis-server")
	cmd.Run()
	defer func() {
		err := cmd.Process.Kill()
		if err != nil {
			fmt.Print(err)
		}
	}()

I know that redis server is already installed on build machine, so this code should run successfully

3. I tried to increase percentage of test coverage, but i faced some troubles in next files:
     cache.go
     config.go

3.1 cache.go. 
To test Bootstrap method i have to had valid DHT (as i understand - this interface has only one implementation - Kademlia). But i can't create Kademlia with not nil dht field. So in the first line of Bootstrap method i throw panic
nodes, err := o.DHT.GetNodes(ctx, "0", 1280)

The same problems with Refresh and Walk methods.

3.2  config.go
the same problem with bootstrap method.

For example, 

 case "redis":
                db, err := strconv.Atoi(dburl.Query().Get("db"))
                if err != nil {
                        return Error.New("invalid db: %s", err)
                }
                cache, err = NewRedisOverlayCache(dburl.Host, urlPwd(dburl), db, kad)
                if err != nil {
                        return err
                }
                zap.S().Info("Starting overlay cache with Redis")

i'm able to create valid cache with calling NewRedisOverlayCache, but my DHT in cache will be broken. 
In this case i will throw panic in next line 
err = cache.Bootstrap(ctx)

**Comment by iglesiasbrandon** [comment]
Going to merge into master since this PR was reviewed and approved by Nadizzle.

**Pull request merged**